### PR TITLE
Sera/issue random sample size

### PR DESCRIPTION
**Pull request by serareif**

### CODE REVIEW

**Review by nictru — CHANGES_REQUESTED** [review]
The problem here is not that the slider is not being updated. The update does happen as expected [here](https://github.com/Mye-InfoBank/SIMBA-QC/blob/e364fe99e6d1130b4f9688b7c22f806f45312482/src/sliders.py#L35). Adding an additional location where the slider is updated introduces unnecessary complexity.

The problem is that the [random sample](https://github.com/Mye-InfoBank/SIMBA-QC/blob/e364fe99e6d1130b4f9688b7c22f806f45312482/src/sliders.py#L77) function will be executed whenever either the slider or the `adata` is updated. 
So the following happens here if the `adata` is updated:
1. `random_sample` and `slider_sample` at the same time
2. `random_sample` again because the slider was updated

The problem will only occur when the `random_sample` is called the first time, as the slider has not been updated yet but the `adata` has

A solution could look as simple as this:
```python
@reactive.effect
def random_sample():
    adata = _adata_qc.get()
    sample_size = input['random_sample_size'].get()
    
    if adata is None:
        return

    adata_sample = adata[np.random.choice(adata.obs.index, min(sample_size, len(adata.obs)), replace=False)]
    _adata_sample.set(adata_sample)
```

**Comment by nictru** [comment]
Also please add a patch release to 0.2.1 [here](https://github.com/Mye-InfoBank/SIMBA-QC/blob/668716eed4dac25dd46bf6329a2703955eb146b9/docker-compose.yml)

File: src/sliders.py [hunk 1]
```diff
@@ -76,12 +76,15 @@ def return_of_adata_meta():
     @reactive.effect
     def random_sample():
         adata = _adata_qc.get()
+        if adata is None:
```
Comment (nictru) [inline src/sliders.py:79]: It should be sufficient to check only once if `adata` is None

**Review by nictru — CHANGES_REQUESTED** [review]

File: src/sliders.py (hunk 1 above)
Comment (serareif) [reply src/sliders.py:79]: okee

Comment (serareif) [reply src/sliders.py:79]: done

**Review by nictru — APPROVED** [review]

**Pull request merged**